packages/react/src/components/ApiKeyManager.tsx

```diff
@@ -24,30 +25,66 @@ const getSystemDefaultThemePreference = (): "dark" | "light" => {
   return "light";
 };
 
-function ApiKeyManager({ provider, menuItems, showIsLoading, theme }: Props) {
-  const queryEngine = useProviderQueryEngine(provider);
-  const query = queryEngine.useMyConsumersQuery();
+const DEFAULT_DATA_MODEL: DataModel = {
+  isFetching: false,
+  consumers: undefined,
+};
+
+function ApiKeyManager({
+  provider,
+  menuItems,
+  theme,
+  enableCreateConsumer,
+  enableDeleteConsumer,
+}: Props) {
   const themeStyle = `zp-key-manager--${
     theme ?? getSystemDefaultThemePreference()
   }`;
+  const [dataModel, setDataModel] = useState<DataModel>(DEFAULT_DATA_MODEL);
+  const [error, setError] = useState<string | undefined>(undefined);
+
+  const loadData = useCallback(async (prov: ApiKeyManagerProvider) => {
+    try {
+      setDataModel({ ...dataModel, isFetching: true });
+      const result = await prov.getConsumers();
+      setDataModel({ consumers: result.data, isFetching: false });
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (err: any) {
+      setError(err.message);
+      console.error(err);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    loadData(provider);
```

Comment:
If you're just trying to make an async call from this you can do
```suggestion
    void loadData(provider);
```